crates/parcel_filesystem/src/lib.rs

```diff
@@ -25,7 +24,7 @@ pub mod os_file_system;
 ///       it should not be in the trait
 /// * [ ] Do not use io results, instead use anyhow or this error
 ///
-pub trait FileSystem {
+pub trait FileSystem: Send + Sync {
```

Comment:
There's no need to force this here; you should enforce on the usage site that the type also is `Send + Sync` if you need it

Reply:
I tried this initially however Rust would not compile with a `Arc<RwLock<Box<dyn FileSystem>>>>` that was sent between futures. I had to at least implement `Sync` - which forced the change to `Mutex`. Either way, better to internalize lock usage imo.

Have `FileSystem: Send + Sync + Clone`

Reply:
You'd want this: `Arc<RwLock<Box<dyn FileSystem + Send>>>>`

Then you don't have to change any of the JS delegate stuff for example

Reply:
We don't need to enforce the send/sync bounds here because it's technically valid to implement the FS API for single-threaded use. 

We can add the Send/Sync bounds where they are needed (e.g.: at the usage)

Reply:
I'll try this

Reply:
Updated